### PR TITLE
Minion v0.4

### PR DESCRIPTION
**Pull request by jrrk**
I generated this pull request for information, however it looks like meld would do a better job ...

### CODE REVIEW

**Comment by wsong83** [comment]
Several issues need to be revised or explained:

1. The PR should be made against master rather than my minion-v0.4-tag. Your changes should not have been pushed to master in the first place. I have it rectified now. So the master branch is reset. Please synchronize your local repo and avoid directly pushing to master again.

2. Is this the finished tutorial for the minion part? Besides the changes of how to run the demo, there is not a single diagram showing how the minion-SD is connected to the original lowRISC SoC (Sorry if I missed this part, please point it out then).

3. There are a lot of pages being just copy and paste from debug-v0.3. I do not think this is the proper way to do it. Two major reasons: 1) the pages related to trace debugger are written by Stefan. Even you put my name on the tutorial, we cannot morally claim authorship if there is no substantial changes being made to those pages. 2) Like code, it is never a good idea to copy something and make redundancy for which we are going to maintain. If a page has no substantial changes, why not reference it in the new tutorial? In this case we only need to change the previous pages if ever needed. We have already done this. See http://www.lowrisc.org/docs/debug-v0.3/lowriscsetup/

4. I see a new branch/release named minion-v0.4-rc1 being made. What is your plan for this branch? What happens if something trivial being pushed to minion-v0.4, then we need to change the tutorial as well (extra maintaining step)?

**Comment by jrrk** [comment]
This tutorial is unfinished, but the walkthrough is updated along with the pre-defined images to check robustness of the instructions. Redundant pages have been removed. Diagrams will be completed last. Thankyou very much for correcting my branches. minion-v0.4-rc1 is a placeholder to contain the binary images release, if it turns out to be satisfactory.

**Comment by wsong83** [comment]
Thanks Jonathan! Now it looks much better. I suppose this PR remains open until it is finished.

**Comment by jrrk** [comment]
The policy of incremental release changes to the tutorial saves duplication, but it does mean users have to refer to old instructions at some points. Then they have to remember to use the browser back button to get back to the current instructions. This approach is satisfactory once users are aware of that, but are there advantages to a self-contained approach?

**Comment by asb** [comment]
I agree with you that there are problems with the current approach - as Wei points out there are problems with duplicating the docs for each release too.

I think the natural endpoint is to have a single set of documentation pages, that are kept up to date (or explicitly marked as retired if they become redundant). There might still be some value in people being able to look back at the docs as they were in e.g. June 2017 but this is more like looking in Wikipedia edit history and shouldn't be something you come across unless you're explicitly looking for it. The best time to move towards continually updated rather than release-specific docs would be when we move to time-based rather than feature-based releases.

**Comment by wsong83** [comment]
to speed it up, I have directly push the changes to minion-v0.4 without a PR. I hope it is fine.

File: content/docs/minion-v0.4/overview.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+move towards that longer-term goal.
+
+In the picture below you can find an updated overview of the lowRISC
+system architecture. If you compare it to the
+[previous SoC overview]({{< ref "docs/debug-v0.3/overview.md" >}}) you
+can see that the major change and main topic of this tutorial is the
+new minion infrastructure.
+
+<a name="figure-overview"></a>
+<img src="../figures/lowRISC_soc.png" alt="Drawing" style="width: 600px; padding: 20px 0px;"/>
```
Comment (wsong83) [inline content/docs/minion-v0.4/overview.md]: This link does not work and lowRISC_soc.png missing.

Comment (jrrk2) [reply content/docs/minion-v0.4/overview.md]: Would the easiest solution be to copy the figure from the debug-v0.3 release ?

**Pull request merged**